### PR TITLE
Ensure User#username and Namespace#name are always valid

### LINKED ISSUE #42: Namespace creation UI
No validation is being done apparently (e.g.: usernames with spaces are accepted).

Docker validates namespace's name using this regexp: `[a-z0-9-_]`

### PR DESCRIPTION
**Pull request by flavio**
The Docker daemon performs some validations for the User's username and
also for the Namespace's name. We have to apply the same policies
inside of Portus.

Fixes #42 and #44

### CODE REVIEW

**Comment by flavio** [comment]
@mssola what do you think about it?

File: app/models/namespace.rb [hunk 1]
```diff
@@ -1,8 +1,15 @@
 class Namespace < ActiveRecord::Base
 
+  NAME_ALLOWED_CHARS = 'a-z0-9\-_'
+
   has_many :repositories
   belongs_to :registry
   belongs_to :team
-  validates :name, presence: true
+  validates :name, presence: true,
+                   format: { with: /\A[#{NAME_ALLOWED_CHARS}]+\Z/, message: 'Only allowed letters: [a-z0-9-_]' }
```
Comment (mssola) [inline app/models/namespace.rb]: Nitpicking: maybe you could write the `message` key on the next line, just like you did in the user.rb file :)

**Comment by mssola** [comment]
LGTM :+1:

**Comment by flavio** [comment]
Fixed, thanks for the feedback

**Pull request merged**